zebra-state/src/service/watch_receiver.rs

```diff
@@ -99,15 +99,25 @@ where
         self.receiver.borrow().clone()
     }
 
-    /// Calls [`watch::Receiver::changed`] and returns the result.
+    /// Calls [`watch::Receiver::changed()`] and returns the result.
+    /// Returns when the inner value has been updated, even if the old and new values are equal.
     ///
     /// Marks the watched data as seen.
     pub async fn changed(&mut self) -> Result<(), watch::error::RecvError> {
         self.receiver.changed().await
     }
 
+    /// Calls [`watch::Receiver::has_changed()`] and returns the result.
+    /// Returns `true` when the inner value has been updated, even if the old and new values are equal.
+    ///
+    /// Does not mark the watched data as seen.
+    pub fn has_changed(&self) -> Result<bool, watch::error::RecvError> {
+        self.receiver.has_changed()
+    }
+
     /// Marks the watched data as seen.
+    /// Calls [`watch::Receiver::mark_changed()`].
     pub fn mark_as_seen(&mut self) {
```

Comment:
Optional: Could we rename this to match the method it calls on `receiver` like `changed()` and `has_changed()`

```suggestion
    pub fn mark_changed(&mut self) {
```

Reply:
This change will be reverted because it's a bug.

We can add a new method if it's ever needed, but we haven't needed it so far.

Reply:
Fixed in 8656076e39, I added the new method anyway for clarity.